src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/Rechargeable.java

```diff
@@ -0,0 +1,145 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import io.github.thebusybiscuit.slimefun4.core.networks.energy.EnergyNet;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.gadgets.Jetpack;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.gadgets.MultiTool;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.ChargingBench;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+/**
+ * A {@link Rechargeable} {@link SlimefunItem} can hold energy and is able to
+ * be recharged using a {@link ChargingBench}.
+ * Any {@link SlimefunItem} which is supposed to be chargeable <b>must</b> implement this interface.
+ * 
+ * @author TheBusyBiscuit
+ * 
+ * @see ChargingBench
+ * @see EnergyNet
+ * @see Jetpack
+ * @see MultiTool
+ *
+ */
+@FunctionalInterface
+public interface Rechargeable extends ItemAttribute {
+
+    /**
+     * This method returns the maximum charge the given {@link ItemStack} is capable of holding.
+     * 
+     * @param item
+     *            The {@link ItemStack} for which to determine the maximum charge
+     * 
+     * @return The maximum energy charge for this {@link ItemStack}
+     */
+    float getMaxItemCharge(ItemStack item);
+
+    /**
+     * This method sets the stored energy charge for a given {@link ItemStack}.
+     * The charge must be at least zero and at most {@link #getMaxItemCharge(ItemStack)}.
+     * 
+     * @param item
+     *            The {@link ItemStack} to charge
+     * @param charge
+     *            The amount of charge to store
+     */
+    default void setItemCharge(ItemStack item, float charge) {
+        if (item == null || item.getType() == Material.AIR) {
+            throw new IllegalArgumentException("Cannot set Item charge for null or AIR");
```

Comment:
How come you never just use Preconditions for these? Bit nicer to look at and generally more consistent with Bukkit internals which I like.

Reply:
Did you ever look at Slimefuns code? I use Validate literally everywhere to check for nulls, but if the expresssions get longer, I stick with an if.

Reply:
Yes I know. Hence why I said "never".